src/sets.jl

```diff
@@ -67,3 +67,18 @@ function LinearAlgebra.norm(::AbstractVector{<:AbstractJuMPScalar})
           "`@constraint(model, norm(x) <= t)` should now be written as ",
           "`@constraint(model, [t; x] in SecondOrderCone())`")
 end
+
+"""
+    Bin
+
+Binary variable category, used in the `@variable` macro.
+"""
+Bin
```

Comment:
I didn't know you could declare variables like this! 

But, maybe we should make it a type? Because it still has this:
```Julia
julia> JuMP.Bin
ERROR: UndefVarError: Bin not defined
```

We also don't really want to declare an `Int` JuMP-local variable. It would get exported and might cause all sorts of issues.

Reply:
It's a bizarre trickery, it's documented, but not an identifier. You get it in REPL auto-completion, but raises UndefVar

Reply:
We can make them types indeed